sites/svelte.dev/scripts/type-gen/index.js

```diff
@@ -219,136 +219,90 @@ function munge_type_element(member, depth = 1) {
-	modules.push({
-		name: 'svelte/animate',
-		comment: '',
-		...get_types(module.code, module.ts_source_file.statements)
-	});
-}
-
-{
-	const module = bundled_types.get('svelte/easing');
-
-	if (!module) throw new Error('Could not find svelte/easing');
-
-	modules.push({
-		name: 'svelte/easing',
-		comment: '',
-		...get_types(module.code, module.ts_source_file.statements)
-	});
+/**
+ * Type declarations include fully qualified URLs so that they become links when
+ * you hover over names in an editor with TypeScript enabled. We need to remove
+ * the origin so that they become root-relative, so that they work in preview
+ * deployments and when developing locally
+ * @param {string} str
+ */
+function strip_origin(str) {
+	return str.replace(/https:\/\/svelte\.dev/g, '');
 }
 
-{
-	const module = bundled_types.get('svelte/motion');
+/**
+ * @param {string} file
+ */
+function read_d_ts_file(file) {
+	const resolved = path.resolve('../../packages/svelte', file);
 
-	if (!module) throw new Error('Could not find svelte/motion');
+	// We can't use JSDoc comments inside JSDoc, so we would get ts(7031) errors if
+	// we didn't ignore this error specifically for `/// file:` code examples
+	const str = fs.readFileSync(resolved, 'utf-8');
 
-	modules.push({
-		name: 'svelte/motion',
-		comment: '',
-		...get_types(module.code, module.ts_source_file.statements)
+	return str.replace(/(\s*\*\s*)```js([\s\S]+?)```/g, (match, prefix, code) => {
+		return `${prefix}\`\`\`js${prefix}// @errors: 7031${code}\`\`\``;
 	});
 }
 
 {
-	const module = bundled_types.get('svelte/store');
+	const code = read_d_ts_file('types/index.d.ts');
+	const node = ts.createSourceFile('index.d.ts', code, ts.ScriptTarget.Latest, true);
 
-	if (!module) throw new Error('Could not find svelte/store');
+	for (const statement of node.statements) {
+		if (ts.isModuleDeclaration(statement)) {
+			// @ts-ignore
+			const name = statement.name.text || statement.name.escapedText;
 
-	modules.push({
-		name: 'svelte/store',
-		comment: '',
-		...get_types(module.code, module.ts_source_file.statements)
-	});
-}
-
-{
-	const module = bundled_types.get('svelte/transition');
+			if (name === '*.svelte' || name === 'svelte/types/compiler/preprocess') {
+				continue;
+			}
 
-	if (!module) throw new Error('Could not find svelte/transition');
+			// @ts-ignore
+			const comment = strip_origin(statement.jsDoc?.[0].comment ?? '');
 
-	modules.push({
-		name: 'svelte/transition',
-		comment: '',
-		...get_types(module.code, module.ts_source_file.statements)
-	});
+			modules.push({
+				name,
+				comment,
+				// @ts-ignore
+				...get_types(code, statement.body?.statements)
+			});
+		}
+	}
 }
 
 modules.sort((a, b) => (a.name < b.name ? -1 : 1));
 
-// Fix the duplicate/messed up types
-// !NOTE: This relies on mutation of `modules`
-$: {
-	const module_with_SvelteComponent = modules.find((m) =>
-		m.types.filter((t) => t.name === 'SvelteComponent')
-	);
+// // Fix the duplicate/messed up types
```

Comment:
should this code be removed now?

Reply:
yeah, i reckon so

Reply:
It was put in place to add JSDoc of SvelteComponentDev to SvelteComponent. Now that Svelte/internal is not part of the d.ts, how do we bring it back? Do we copy paste the stuff of SvelteComponentDev to this?